File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/typename_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'typename'

f = Typename::Foo.new
b = Typename::Bar.new

x = Typename.twoFoo(f)
unless x.is_a? Float
  puts "Wrong return type!"
end

y = Typename.twoBar(b)
unless y.is_a? Integer
  puts "Wrong return type!"
end

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/typename_runme.py
import typename
import types
f = typename.Foo()
b = typename.Bar()

x = typename.twoFoo(f)
if not isinstance(x, types.FloatType):
    raise RuntimeError, "Wrong return type (FloatType) !"
y = typename.twoBar(b)
if not isinstance(y, types.IntType):
    raise RuntimeError, "Wrong return type (IntType)!"
